Docs build broken

Current builds in e.g. #2700 broken with error
```
PYTHONPATH=/home/runner/work/tskit/tskit/docs/../python/ ./build.sh
280
./build.sh: 9: jupyter-book: not found
281
make: *** [Makefile:16: dev] Error 127
282
make: Leaving directory '/home/runner/work/tskit/tskit/docs'
```
Perhaps a new version of jupyter book has popped up, or we're not installing the binary into the PATH anymore or something?

Would you mind taking a look please @benjeffery?

--- a/.github/workflows/docs.yml
+++ b/.github/workflows/docs.yml
@@ -40,7 +40,7 @@ jobs:
         id: venv-cache
         with:
           path: venv
-          key: docs-venv-v1-${{ hashFiles(env.REQUIREMENTS) }}
+          key: docs-venv-v2-${{ hashFiles(env.REQUIREMENTS) }}
 
       - name: Create venv and install deps (one by one to avoid conflict errors)
         if: steps.venv-cache.outputs.cache-hit != 'true'

